SC_redesign/Crop_and_Soil/field/fertilizer_application.py

```
@@ -0,0 +1,65 @@
+from typing import Optional
+
+from SC_redesign.Crop_and_Soil.soil.soil import Soil
+
+"""
+This module provides a way for Field to apply fertilizer, based on SWAT Theoretical documentation section 6:1.7.
+"""
+
+
+class FertilizerApplication:
+
+    def __init__(self, soil: Optional[Soil] = None, field_size: Optional[float] = None):
```

This part seems a bit tricky. Ideally, we want exactly one argument. When no arguments are provided, should we throw an error? But when both are provided, the field size just becomes redundant. I think the confusion comes from having two optional parameters but at least one is actually required. 

I think the better design is to require a soil object since the client code should have either a soil object or know the field size. For the latter, it may as well create a soil object anyway.

---

You make a good point here - one argument would be a lot easier to understand.

Some time ago, there was only one optional input - `Soil`, and if not provided would create a new `Soil` object to work with. Then some features got added, and lead to the necessity for the field size to be provided when creating new `SoilData` objects, which is why this module (and all others in the `soil/` submodule) have two optional initialization parameters.

If there is an attempt to instantiate a `FertilizerApplication` object with `soil` and `field_size` both being `None`, an error will be raised. But it is raised in the `post_init()` of `SoilData` which seems not ideal.

The idea behind making the `Soil` (or in other modules, `SoilData`) optional was that if a user or researcher was interested in using or studying the behavior of only one module, this design would require them to do as little overhead setup as possible.

We should talk more about what a better input/initialization system would be, but for now I think it should be left like this because it is consistent with the rest of the Crop and Soil module.